### ISSUE TITLE
pppwn list interface (exclude Bluetooth Device, Virtual Adapter, Wi-Fi)

### ISSUE THREAD

**Issue by oresterosso75** [xfangfang/PPPwn_cpp#40]
![pppwn](https://github.com/xfangfang/PPPwn_cpp/assets/50875656/c9a1d50d-093c-49a8-a051-ff851e05e8c4)
I noticed that pppwn list returns a list of all the interfaces present in the device.
non-essential ones such as Bluetooth Device, Virtual Adapter, Wi-Fi should be excluded from the search.
who knows, maybe filtering with regex?

**Comment by xfangfang** [comment]
I use the api provided by the system to obtain network interfaces under macOS. I believe similar functions are also available on Windows. Feel welcome to submit a PR.

https://github.com/xfangfang/PPPwn_cpp/blob/5c0b15a8bfb9bcf9108239406e0573fe3aa808cb/src/main.cpp#L78-L111

**Comment by Gouster4** [comment]
Im not sure about bluetooth, but i guess that ESP32 or similar microcontroller can act as Bluetooth to Wifi bridge, so maybe even bluetooth connection would be possible. I bet that no one would do it over bluetooth, but its still an option.

**Comment by annahana** [comment]
> Im not sure about bluetooth, but i guess that ESP32 or similar microcontroller can act as Bluetooth PAN to Wifi bridge, so maybe even bluetooth connection would be possible. I bet that no one would do it over bluetooth, but its still an option. Maybe also some virtual adapters can be used for PPPwn, like tunnel interfaces that on other end are bridged to router/firewall port which is connected to PS4. But dont take my word on it, i did never tested such a scenario.

Yes possible to do it like that. esp32 is not likely to work(much to slow for Bluetooth and wifi)..

**Comment by Gouster4** [comment]
So, its reasonable to leave all those options available, because all of them can be used, just not very attractive as LAN or Wifi, and most likely no one would use them. But its still an possible option.

**Comment by annahana** [comment]
Real hardwar should be listet but the virtual ones and bridges should not be listet. Nobody will ever send the jailbreak to a VM or the bridge between lan and WLAN.adapter.

**Comment by Gouster4** [comment]
> Real hardwar should be listet but the virtual ones and bridges should not be listet. Nobody will ever send the jailbreak to a VM or the bridge between lan and WLAN.adapter.
When you bridge two interfaces, isn't then bridge itself that interface which you use to communicate with those interfaces? At least I do that so on my firewall.

**Comment by oresterosso75** [comment]
in my form I use this function:

`private void ethernetScan()
{
    NetworkInterface[] schedeDiRete = NetworkInterface.GetAllNetworkInterfaces();

    string pattern = @"(\bwi-fi\b|\bwifi\b|\bwireless\b|\bbluetooth\b)";

    foreach (NetworkInterface scheda in schedeDiRete)
    {
        string nomeScheda = scheda.Description.ToLower();

        // Escludi le schede di rete virtuali come la loopback
        if (!Regex.IsMatch(nomeScheda, pattern) && !scheda.Description.ToLower().Contains("loopback"))
        {
            comboBox2.Items.Add(new SchedaDiReteItem { Scheda = scheda, Descrizione = scheda.Description });
        }
    }
}
`
### Retrieval of Network Interfaces
`NetworkInterface.GetAllNetworkInterfaces()` retrieves all the network interfaces available on the system.

### Exclusion Pattern
The pattern `@"(\bwi-fi\b|\bwifi\b|\bwireless\b|\bbluetooth\b)"` is used to exclude network interfaces based on Wi-Fi, wireless, and Bluetooth.

### `foreach` Loop
Iterates over each network interface.

### Interface Name
`scheda.Description.ToLower()` converts the interface description to lowercase to facilitate comparison.

### Exclusion Condition
Network interfaces that contain the terms from the pattern or "loopback" in their description are excluded.

### Adding to ComboBox
Network interfaces that are not excluded are added to `comboBox2` as `SchedaDiReteItem` objects.

**Comment by annahana** [comment]
> > Real hardwar should be listet but the virtual ones and bridges should not be listet. Nobody will ever send the jailbreak to a VM or the bridge between lan and WLAN.adapter.
> 
> When you bridge two interfaces, isn't then bridge itself that interface which you use to communicate with those interfaces? At least I do that so on my firewall. I'm not an expert, so maybe I'm doing it wrong.

No technically it is a network on its one. If you use it as a interface you send the jealbreak to both sides wlan and lan. In best case you just DDoS your Home Network.

**Comment by Gouster4** [comment]
> No technically it is a network on its one

Hm... so its better to send it directly to specific interface. Got it.
